test/format/ops.c

```diff
@@ -878,25 +958,59 @@ ops(void *arg)
      * pound on the same key/value pairs, that is, by making them traverse the same RNG space. 75%
      * of the time we run in independent RNG space.
      */
-    if (GV(FORMAT_INDEPENDENT_THREAD_RNG))
-        __wt_random_init_seed(NULL, &tinfo->rnd);
-    else
-        __wt_random_init(&tinfo->rnd);
+    if (GV(FORMAT_INDEPENDENT_THREAD_RNG)) {
+        __wt_random_init_seed(NULL, &tinfo->data_rnd);
+        __wt_random_init_seed(NULL, &tinfo->extra_rnd);
+    } else {
+        __wt_random_init(&tinfo->data_rnd);
+        __wt_random_init(&tinfo->extra_rnd);
+    }
 
     iso_level = ISOLATION_SNAPSHOT; /* -Wconditional-uninitialized */
+    tinfo->replay_again = false;
+    tinfo->lane = LANE_NONE;
 
     /* Set the first operation where we'll create a new session and cursors. */
     session = NULL;
     session_op = 0;
+    ntries = 0;
 
     /* Set the first operation where we'll reset the session. */
-    reset_op = mmrand(&tinfo->rnd, 100, 10 * WT_THOUSAND);
+    reset_op = mmrand(&tinfo->extra_rnd, 100, 10 * WT_THOUSAND);
     /* Set the first operation where we'll truncate a range. */
-    truncate_op = mmrand(&tinfo->rnd, 100, 10 * WT_THOUSAND);
+    truncate_op = mmrand(&tinfo->data_rnd, 100, 10 * WT_THOUSAND);
 
     for (intxn = false; !tinfo->quit;) {
+rollback_retry:
+        if (tinfo->quit)
+            break;
+
         ++tinfo->ops;
 
+        if (!tinfo->replay_again)
+            /*
+             * Number of failures so far for the current operation and key. In predictable replay,
+             * unless we have a read operation, we cannot give up on any operation and maintain the
+             * integrity of the replay.
+             */
+            ntries = 0;
+
+        /* Number of tries only gets incremented during predictable replay. */
+        testutil_assert(ntries == 0 || (!intxn && tinfo->replay_again));
+
+        /*
+         * In predictable replay, put each operation in its own transaction. It's possible we could
+         * make multiple operations work predictably in the future.
```

Comment:
Does it really seem possible? I.e., is it reasonable to create a ticket so maybe we get to it in the future?

Reply:
Yes, it seems possible - I'll open a ticket for discussion.

Reply:
I put this into WT-10571 with some thoughts, it doesn't seem too hard.  It's in the followon Epic.